task_4/task4.py

```diff
@@ -0,0 +1,34 @@
+import os
+import pyarrow.parquet as pq
+import requests
+from tqdm import tqdm
+
+# Load the parquet file
+table = pq.read_table('links.parquet')
+
+# Get the first 10,000 URLs
+urls = table['URL'][:10000]
+
+# Create a directory to store the images
+os.makedirs('images', exist_ok=True)
+
+
+# Define headers
+headers = {
+    'User-Agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/58.0.3029.110 Safari/537.3'
+}
+
+for i, url in enumerate(tqdm(urls, desc='Downloading images')):
+    try:
+        response = requests.get(url, headers=headers, stream=True)
+        response.raise_for_status()
+
+        # Save the image to a file
+        with open(f'images/image_{i}.jpg', 'wb') as f:
+            for chunk in response.iter_content(chunk_size=8192):
```

Comment:
why chunking?

Reply:
incase of large file to limit memory usage

Reply:
👍